File: Tests.Common.VB/CompilerGenerated/Literals.vb
Imports System.Linq.Expressions.Expression

Partial Public Class VBCompilerGeneratedBase
    <Fact> <Trait("Category", Literal)>
    Sub [True]()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub [False]()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub NothingString()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub [Nothing]()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub [Integer]()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub NonInteger()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub [String]()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub DateTime()
        Dim expr = Functions.Expr(Function() #1981-1-1#)
        Assert.Equal($"Function() #{#1981-1-1#.ToString()}#", expr.ToString("Visual Basic"))
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub EscapedString()
        PreRunTest()
    End Sub

    <Fact> <Trait("Category", Literal)>
    Sub InterpolatedString()
        Dim expr = Functions.Expr(Function() $"{#2001-1-1#}")
        Assert.Equal("Function() $""{#" + #2001-1-1#.ToString + "#}""", expr.ToString("Visual Basic"))
    End Sub
End Class

File: Tests.Common/CompilerGenerated/Literals.cs
using Xunit;
using static ExpressionToString.Tests.Categories;

namespace ExpressionToString.Tests {
    public partial class CompilerGeneratedBase {
        [Fact]
        [Trait("Category", Literal)]
        public void True() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void False() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void Nothing() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void Integer() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void NonInteger() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void String() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void InterpolatedString() => PreRunTest();

        [Fact]
        [Trait("Category", Literal)]
        public void Type() => PreRunTest();
    }
}
